homeassistant/components/overkiz/water_heater/atlantic_domestic_hot_water_production_mlb_component.py

```diff
@@ -167,13 +168,19 @@ async def async_turn_away_mode_on(self) -> None:
         With `refresh_afterwards=False` on the first commands, and `refresh_afterwards=True` only the last command,
         the API is not choking and the transition is smooth without the unavailability state.
         """
-        now_date = cast(
-            dict,
-            self.executor.select_state(OverkizState.CORE_DATETIME),
-        )
+        now = dt_util.now()
+        now_date = {
+            "month": now.month,
+            "hour": now.hour,
+            "year": now.year,
+            "weekday": now.weekday(),
+            "day": now.day,
+            "minute": now.minute,
+            "second": now.second,
+        }
         await self.executor.async_execute_command(
-            OverkizCommand.SET_ABSENCE_MODE,
-            OverkizCommandParam.PROG,
+            OverkizCommand.SET_DATE_TIME,
```

Comment:
Do you need to set this date every command, or can you do this in the constructor, so that it will be done every reboot.

Will this date be reset automatically?

Reply:
I cannot say for sure when the datetime gets reset, I have no statistics on it, but it occurs regularly, and it needs only one mismatch for the mode not to be able to be turned on until the datetime is in sync again.
A singleton may do, but what if the user runs their HA for a month? a year without reboots?
Yes, it may be overhead, but it makes the operation successful every time instead of confusing the user, and, what's even worse, instead of the away mode being randomly turned off, which can cause huge money loss for a user.

Reply:
The device seems to lose the correct time after switching to absence mode. It appears that the time needs to be synchronized each time absence mode is activated. Without this my device doesn’t seem to function reliably. In general, the device seems to lose the time without any specific reason. Many people use absence mode to avoid the high-cost hours of electricity, so the feature should work reliably.